tests/Backend/Bigquery/TableDefinitionOperationsPartitioningTest.php

```diff
@@ -0,0 +1,218 @@
+        ]);
+
+        $tableResponse = $this->_client->getTable($tableId);
+        $expectedTableDefinition = [
+            'primaryKeysNames' => [
+                'id',
+            ],
+            'columns' => [
+                [
+                    'name' => 'id',
+                    'definition' => [
+                        'type' => 'INTEGER',
+                        'nullable' => false,
+                    ],
+                    'basetype' => 'INTEGER',
+                    'canBeFiltered' => true,
+                ],
+                [
+                    'name' => 'time',
+                    'definition' => [
+                        'type' => 'TIMESTAMP',
+                        'nullable' => false,
+                    ],
+                    'basetype' => 'TIMESTAMP',
+                    'canBeFiltered' => true,
+                ],
+            ],
+            'timePartitioning' => [
+                'type' => 'DAY',
+                'field' => 'time',
+                'expirationMs' => '1000',
+            ],
+            'clustering' => [
+                'fields' => [
+                    'id',
+                ],
+            ],
+            'requirePartitionFilter' => false,
+            'partitions' => [],
+        ];
+        $this->assertSame($expectedTableDefinition, $tableResponse['definition']);
+
+        $csvFile = $this->createTempCsv();
+        $csvFile->writeRow([
+            'id',
+            'time',
+        ]);
+        $csvFile->writeRow([
+            '1',
+            '2020-01-01 00:00:00',
+        ]);
+        $this->_client->writeTableAsync($tableId, $csvFile);
+
+        $tableResponse = $this->_client->getTable($tableId);
+        $this->assertSame(
+            array_merge(
+                $expectedTableDefinition,
+                [
+                    'partitions' => [
+                        // todo: expected one partition https://keboola.atlassian.net/browse/BIG-186
+                        // same apply for partitions after snapshot restore
+                    ],
+                ]
+            ),
+            $tableResponse['definition']
+        );
+
+        // test snapshots
+        $snapshotId = $this->_client->createTableSnapshot($tableId);
+        $newTableId = $this->_client->createTableFromSnapshot($tableResponse['bucket']['id'], $snapshotId, 'restored');
+        $newTable = $this->_client->getTable($newTableId);
+        $this->assertSame($expectedTableDefinition, $newTable['definition']);
+
+        $workspace = $this->initTestWorkspace();
+        // test workspace load
+        // if we would implement copy load this needs to be also tested
+//        $workspaces = new Workspaces($this->_client);
+//        $workspaces->loadWorkspaceData($workspace['id'], [
+//            'input' => [
+//                [
+//                    'source' => $tableId,
+//                    'destination' => 'partitioned',
+//                ],
+//            ],
+//        ]);
```

Comment:
Pridal by som to ako try catch a testoval ze to naozaj spadne, aby ked potom pridame podporu tak sme nezabudli na toto miesto a spadol test ze to mame upravit.

Reply:
No to je složitější on nespadne :D ale zkusím něco vymyslet :D